# Cleaning the Data and Exploring 

This part explores the data, and apply the following methods to ensure the integrity of it:
- **Handle Missing Values:** Techniques include deletion, imputation (filling with mean, median, or mode), or using algorithms that can handle missing data.
- **Correct Inaccuracies:** Identify incorrect data points and correct them manually or automatically.
- **Standardize Data:** Ensure consistency in data formats (e.g., dates, phone numbers).
- **Remove Irrelevant Data:** Exclude data that is not necessary for the analysis to reduce noise.

In [1]:
import pandas as pd
import numpy as np
# import matplotlib.pyplot as plt

In [2]:
df = pd.read_csv("oral-health-uofg.csv")
df.head()

طابع زمني Are you a student at University of Gezira? Age Gender  \
0  9/24/2023 22:01:24                                        Yes  23   Male   
1  9/24/2023 22:04:16                                        Yes  25   Male   
2  9/24/2023 22:05:24                                        Yes  24   Male   
3  9/24/2023 22:09:19                                        Yes  25   Male   
4  9/24/2023 22:09:26                                        Yes  20   Male   

     Faculty Academic level  \
0   Medicine    Fourth year   
1   Medicine     Fifth year   
2  Dentistry    Fourth year   
3   Medicine     Fifth year   
4   Medicine     Third year   

                             Causes of dental caries  \
0  Toothpaste without fluoride, Frequent ingestio...   
1                        Dynamic of oral micro-flora   
2  Frequent ingestion of sugar, Dynamic of oral m...   
3  Frequent ingestion of sugar, Inadequate tooth ...   
4  Frequent ingestion of sugar, Dynamic of oral m...   

           Causes of bleeding during tooth brushing:  \
0                                  Brushing too hard   
1                                Periodontal disease   
2  Natural physiological phenomenon, Brushing too...   
3  Periodontal disease, Brushing too hard, System...   
4                Brushing too hard, Systemic disease   

                         Influence of dental plaque:  \
0                               Affecting appearance   
1                               Affecting appearance   
2                                         Don't know   
3  Affecting appearance, Inducing dental caries, ...   
4       Affecting appearance, Inducing dental caries   

             Measures that prevent oral diseases:  \
0                         Application of fluoride   
1                         Application of fluoride   
2  Application of fluoride, Pit & fissure sealing   
3          Application of fluoride, Tooth scaling   
4                         Application of fluoride   

  Systemic diseases that may be related to oral diseases:  \
0                                     Heart diseases        
1                                     Heart diseases        
2                  None of the above, Other diseases        
3    Heart diseases, Diabetes mellitus, Hypertension        
4                                  None of the above        

  Which is more important for oral health: self-administration or dentist?:  \
0                Self-administration of oral hygiene                          
1                Self-administration of oral hygiene                          
2                           Regular visit to dentist                          
3                Self-administration of oral hygiene                          
4                           Regular visit to dentist                          

  Frequency of daily tooth brushing. Duration of tooth brushing.  \
0                                  1              2 to 3 minutes   
1                                  2         More than 3 minutes   
2                                  2              2 to 3 minutes   
3                                  1         More than 3 minutes   
4                                  1              2 to 3 minutes   

  Frequency of replacing tooth brush.  \
0                  Every three months   
1                  Every three months   
2                                 NaN   
3                  Every three months   
4              Until it can't be used   

              Frequency of visiting dentist. Method of tooth brushing:  \
0  When I have a dental problem/oral disease                 Irregular   
1                               Twice a year          Horizontal scrub   
2                               Twice a year   Modified pass technique   
3                                Once a year            Vertical scrub   
4  When I have a dental problem/oral disease          Horizontal scrub   

  Oral hygiene methods besides tooth brushing:  \
0                          

In [3]:
df.reset_index(inplace=True)
df.rename(columns={"index": "RecordID"}, inplace=True)
df.head()

RecordID           طابع زمني Are you a student at University of Gezira?  \
0         0  9/24/2023 22:01:24                                        Yes   
1         1  9/24/2023 22:04:16                                        Yes   
2         2  9/24/2023 22:05:24                                        Yes   
3         3  9/24/2023 22:09:19                                        Yes   
4         4  9/24/2023 22:09:26                                        Yes   

  Age Gender    Faculty Academic level  \
0  23   Male   Medicine    Fourth year   
1  25   Male   Medicine     Fifth year   
2  24   Male  Dentistry    Fourth year   
3  25   Male   Medicine     Fifth year   
4  20   Male   Medicine     Third year   

                             Causes of dental caries  \
0  Toothpaste without fluoride, Frequent ingestio...   
1                        Dynamic of oral micro-flora   
2  Frequent ingestion of sugar, Dynamic of oral m...   
3  Frequent ingestion of sugar, Inadequate tooth ...   
4  Frequent ingestion of sugar, Dynamic of oral m...   

           Causes of bleeding during tooth brushing:  \
0                                  Brushing too hard   
1                                Periodontal disease   
2  Natural physiological phenomenon, Brushing too...   
3  Periodontal disease, Brushing too hard, System...   
4                Brushing too hard, Systemic disease   

                         Influence of dental plaque:  \
0                               Affecting appearance   
1                               Affecting appearance   
2                                         Don't know   
3  Affecting appearance, Inducing dental caries, ...   
4       Affecting appearance, Inducing dental caries   

             Measures that prevent oral diseases:  \
0                         Application of fluoride   
1                         Application of fluoride   
2  Application of fluoride, Pit & fissure sealing   
3          Application of fluoride, Tooth scaling   
4                         Application of fluoride   

  Systemic diseases that may be related to oral diseases:  \
0                                     Heart diseases        
1                                     Heart diseases        
2                  None of the above, Other diseases        
3    Heart diseases, Diabetes mellitus, Hypertension        
4                                  None of the above        

  Which is more important for oral health: self-administration or dentist?:  \
0                Self-administration of oral hygiene                          
1                Self-administration of oral hygiene                          
2                           Regular visit to dentist                          
3                Self-administration of oral hygiene                          
4                           Regular visit to dentist                          

  Frequency of daily tooth brushing. Duration of tooth brushing.  \
0                                  1              2 to 3 minutes   
1                                  2         More than 3 minutes   
2                                  2              2 to 3 minutes   
3                                  1         More than 3 minutes   
4                                  1              2 to 3 minutes   

  Frequency of replacing tooth brush.  \
0                  Every three months   
1                  Every three months   
2                                 NaN   
3                  Every three months   
4              Until it can't be used   

              Frequency of visiting dentist. Method of tooth brushing:  \
0  When I have a dental problem/oral disease                 Irregular   
1                               Twice a year          Horizontal scrub   
2                               Twice a year   Modified pass technique   
3                                Once a year            Vertical scrub   
4  When I have a dental problem/oral disease          Horizontal scrub   

  Oral h

In [4]:
df = df[
    df["Are you a student at University of Gezira?"] == "Yes"
]  # Filtering all students who are NOT students at UofG

In [5]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 401 entries, 0 to 407
Data columns (total 20 columns):
 #   Column                                                                     Non-Null Count  Dtype 
---  ------                                                                     --------------  ----- 
 0   RecordID                                                                   401 non-null    int64 
 1   طابع زمني                                                                  401 non-null    object
 2   Are you a student at University of Gezira?                                 401 non-null    object
 3   Age                                                                        396 non-null    object
 4   Gender                                                                     400 non-null    object
 5   Faculty                                                                    401 non-null    object
 6   Academic level                                                         

Most columns that are missing are considerably low in number which can handle filling it with the mode. Except for "Oral hygiene methods besides tooth brushing" which probably indicate "I use nothing else" so this will be filled with "None".

In [6]:
def clean_age(val):
    try:
        return int(val)  # works if val is already a number or numeric string
    except:
        try:
            return int(val[:2])
        except:
            return np.nan


df["Age"] = df["Age"].apply(clean_age)
print(df["Age"])

0      23.0
1      25.0
2      24.0
3      25.0
4      20.0
       ... 
403    24.0
404    24.0
405    25.0
406    25.0
407    24.0
Name: Age, Length: 401, dtype: float64


In [7]:
df.isna().sum()

RecordID                                                                      0
طابع زمني                                                                     0
Are you a student at University of Gezira?                                    0
Age                                                                           6
Gender                                                                        1
Faculty                                                                       0
Academic level                                                                1
Causes of dental caries                                                       3
Causes of bleeding during tooth brushing:                                     5
Influence of dental plaque:                                                   8
Measures that prevent oral diseases:                                          8
Systemic diseases that may be related to oral diseases:                       7
Which is more important for oral health:

In [8]:
import statistics

for col in df.columns:
    if df[col].isna().sum() > 0 and df[col].isna().sum() < 20:
        df.fillna({col: statistics.mode(df[col])}, inplace=True)

In [9]:
df.isna().sum()

RecordID                                                                      0
طابع زمني                                                                     0
Are you a student at University of Gezira?                                    0
Age                                                                           0
Gender                                                                        0
Faculty                                                                       0
Academic level                                                                0
Causes of dental caries                                                       0
Causes of bleeding during tooth brushing:                                     0
Influence of dental plaque:                                                   0
Measures that prevent oral diseases:                                          0
Systemic diseases that may be related to oral diseases:                       0
Which is more important for oral health:

In [10]:
df.fillna({"Oral hygiene methods besides tooth brushing:": "None"}, inplace=True)
df.isna().sum()  # cleaning is done

RecordID                                                                     0
طابع زمني                                                                    0
Are you a student at University of Gezira?                                   0
Age                                                                          0
Gender                                                                       0
Faculty                                                                      0
Academic level                                                               0
Causes of dental caries                                                      0
Causes of bleeding during tooth brushing:                                    0
Influence of dental plaque:                                                  0
Measures that prevent oral diseases:                                         0
Systemic diseases that may be related to oral diseases:                      0
Which is more important for oral health: self-admini

In [11]:
df.rename(columns={"طابع زمني": "Timestamp"}, inplace=True)

In [12]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 401 entries, 0 to 407
Data columns (total 20 columns):
 #   Column                                                                     Non-Null Count  Dtype  
---  ------                                                                     --------------  -----  
 0   RecordID                                                                   401 non-null    int64  
 1   Timestamp                                                                  401 non-null    object 
 2   Are you a student at University of Gezira?                                 401 non-null    object 
 3   Age                                                                        401 non-null    float64
 4   Gender                                                                     401 non-null    object 
 5   Faculty                                                                    401 non-null    object 
 6   Academic level                                                 

In [13]:
df["Academic level"].value_counts()

Academic level
Third year     108
Fifth year      92
Fourth year     86
Second year     61
First year      54
Name: count, dtype: int64

Applying the inclusion and exclusion criteria reduces the sample size to dangerous levels to 254 from 401! This threatens the following:

- Statistical power (increased margin of error)
- Generalizability
- Validity of between-group comparisons

All Academic levels will be kept for now.

Note to researchers: Just be transparent about this change in your methodology section and explain it as a tradeoff to preserve sample integrity.

In [14]:
df["Gender"].value_counts()

Gender
Female    306
Male       95
Name: count, dtype: int64

In [15]:
df["Faculty"].value_counts()

Faculty
Medicine     341
Dentistry     60
Name: count, dtype: int64

There's 76% female and 24% male distribution and 85% medicine, 15% dentistry.

The study should account for the limitations of sampling that may also introduce bias. Consequently, the findings, particularly those related to gender or faculty specific trends should be interpreted with caution.

The smaller representation of dentistry students, for instance, may limit the generalizability of any comparisons made between faculties. Future studies are encouraged to adopt more stratified or randomized sampling approaches to ensure more balanced representation.

In [16]:
df.to_csv("oral-health-uofg-cleaned.csv", index=False)